server/routes/api/landingpage.py

```diff
@@ -0,0 +1,292 @@
+                    [dcid] + cache_data.get('similarPlaces', []))
+                # Nearby places data
+                chart['nearby'] = get_bar_data(
+                    chart, cache_data['data'],
+                    [dcid] + cache_data.get('nearbyPlaces', []))
+                # Child places data
+                chart['child'] = get_bar_data(chart, cache_data['data'],
+                                              [dcid] +
+                                              cache_data.get('childPlaces', []))
+                if (chart['trend'] or chart['parent'] or chart['similar'] or
+                        chart['nearby'] or chart['child']):
+                    child_charts.append(chart)
+            if child_charts:
+                child['charts'] = child_charts
+                filtered_children.append(child)
+        if filtered_children:
+            category['children'] = filtered_children
+
+    allPlaces = [dcid]
+    for relation in [
+            'parentPlaces', 'similarPlaces', 'nearbyPlaces', 'childPlaces'
+    ]:
+        allPlaces.extend(cache_data.get(relation, []))
+
+    names = place_api.get_display_name('^'.join(sorted(allPlaces)))
+
+    response = {
+        'configData': config_data,
+        'allChildPlaces': cache_data.get('allChildPlaces', {}),
+        'childPlaces': cache_data.get('childPlaces', []),
+        'parentPlaces': cache_data.get('parentPlaces', []),
+        'similarPlaces': cache_data.get('similarPlaces', []),
+        'nearbyPlaces': cache_data.get('nearbyPlaces', []),
+        'names': names,
+    }
+    return Response(json.dumps(response), 200, mimetype='application/json')
+
+
+def get_bar_data(cc, data, places):
+    """Get the bar data across a few places.
+
+    This will scale the value if required and pick the latest date that has the
+    most <place, stat_var> entries.
+    """
+    if not places:
+        return {}
+    date_to_data = collections.defaultdict(
+        lambda: collections.defaultdict(list))
+
+    denominator_stat_var = None
+    if 'relatedChart' in cc and cc['relatedChart'].get('scale', False):
+        denominator_stat_var = cc['relatedChart'].get('denominator',
+                                                      'Count_Person')
+
+    sources = set()
+    for dcid in places:
+        if dcid not in data:
+            continue
+        for i, stat_var in enumerate(cc['statsVars']):
+            series_raw = data[dcid].get(stat_var, {})
+            if not series_raw:
+                continue
+            if 'denominator' in cc:
+                denominator_stat_var = cc['denominator'][i]
+            if denominator_stat_var is not None:
+                denominator_raw = data[dcid].get(denominator_stat_var, {})
+                if not denominator_raw:
+                    continue
+                series = scale_series(series_raw['data'],
+                                      denominator_raw['data'])
+                sources.add(series_raw['provenanceDomain'])
```

Comment:
could you add a note that we should only add this to sources if we end up picking the place for the final result (this could happen with international place comparisons)